src/vs/workbench/contrib/positronPlots/browser/positronPlotsView.tsx

```diff
@@ -206,8 +195,31 @@ export class PositronPlotsViewPane extends PositronViewPane implements IReactCom
 		this._positronPlotsContainer = DOM.$('.positron-plots-container');
 		container.appendChild(this._positronPlotsContainer);
 
+		// Observe the plots container for resizes and fire size/position changed events.
+		// This is needed in addition to the layoutBody override to trigger React renders
+		// when either the plots pane or a neighboring pane is expanded/collapsed,
+		// since the expand/collapse transition may be animated. Otherwise, the size/position
+		// changed events would only fire at the beginning of the animation possibly leading
+		// to incorrect layouts.
+		this._positronPlotsContainerResizeObserver = new ResizeObserver(entries => {
```

Comment:
Is there a chance the observer could already exist here from a previous call to `renderBody`? If so, do we need to disconnect it before overwriting it to avoid accumulating observers?

Reply:
Good point, addressed in ae5dfdd.